distributed/worker.py

```diff
@@ -3422,7 +3433,12 @@ def done_event():
                 who_has = await retry_operation(
                     self.scheduler.who_has, keys=refresh_who_has
                 )
-                self.update_who_has(who_has)
+                refresh_stimulus_id = f"refresh-who-has-{time()}"
```

Comment:
It would, but atm we don't have a design for tracking stimulus IDs that cause other stimulus IDs. We definitely don't want the same, or we'd end up with the same stimulus ID throughout the whole task graph potentially (I went through this design issue for compute task too). Maybe we could add a log event?
```python
self.log(('stimulus-id-trigger', refresh_stimulus_id, stimulus_id, time()))
```
There are several other places where this design could be applied so I'd rather leave it to a separate PR.